types/block.go

```diff
@@ -5,15 +5,23 @@ import (
 	"github.com/make-software/casper-go-sdk/types/keypair"
 )
 
-// Block in the network
+type BlockWithSignatures struct {
+	Block  Block   `json:"block"`
+	Proofs []Proof `json:"proofs"`
+}
+
 type Block struct {
-	Hash   key.Hash    `json:"hash"`
-	Header BlockHeader `json:"header"`
-	Body   BlockBody   `json:"body"`
-	Proofs []Proof     `json:"proofs"`
+	BlockV1 *BlockV1 `json:"Version1"`
```

Comment:
If we do so, it breaks all clients using the Block structure, father more the fields of this structure contains also different types, so clients should be rewritten significantly.
```
var eraID uint32
if (s.switchBlock.BlockV1 != nil) {
eraID = s.switchBlock.BlockV1.Header.EraID 
} else {
eraID = s.switchBlock.BlockV1.Header.EraID
}
```
and so on for each property....
Don't we want to come up with other ideas that will save customers from this pain?

Reply:
Are you sure that block version is important for most cases? I'm worrying about the smooth transfer of SDK clients to v0.2. With the current design when v2.0 is released, they can't use it with the old SDK version as well and they won't easily updates to the new version without significant code changes. The process should take dev time, losing data and following migration problems. I suggest thinking about a possibility to simplify this process for clients.

Reply:
I took the same approach for .NET SDK. And, as far as we know, Casper took also the same decision with Javascript SDK. SDKs for v1.x are not working on v2.x, and viceversa.